File: stack/stack.py
class Stack:
    def __init__(self, stack = []):
        self.stack = stack

    def pop(self):
        if len(self.stack) > 0:
            return self.stack.pop()
        return None

    def push(self, value):
        self.stack.append(value)


    def peek(self):
        if len(self.stack) > 0:
            return self.stack[len(self.stack) - 1]
        return None

File: stack/stack.cpp
#include <iostream>
#include <string>

using namespace std;

template<typename T, int length>
class Stack{
    private:
    T stack[length];
    int top, count;

    public:
    bool isStackEmpty() const { return top == -1; }
    bool isStackFull(){return count == length;};

    void push(T& value){
        try{
            if(isStackFull()){
                throw string("stack is full");
            }else{
                top++;
        count++;
        this->stack[top] = value;
            }
        }catch(string exception){
            cout << exception << ::endl;
        }
        
    }

    T & pop(){
        try{
            if(isStackEmpty())
                throw string("stack empty");
            else{
                return this->stack[top];
                top--;
                count--;
            }
        }catch(string error){
            cout << error << ::endl;
        }
    }

   

    Stack(): top(-1), count(0){};
};

int main(){
    Stack<int, 5> stack;
    stack.pop();

    return 0;
}